devices/AtomQrCode/QrCodeReader.cs

```diff
@@ -0,0 +1,843 @@
+        private byte[] _readBuffer;
+        private int _readBufferIndex;
+
+        private bool _disposed;
+
+        private TimeSpan _lastTimeout;
+
+        // backing fields for properties
+        private readonly string _portName;
+        private readonly SerialPort _readerSerialPort;
+        private bool _waitingForData = false;
+        private TriggerMode _triggerMode = TriggerMode.None;
+
+        /// <summary>
+        /// Get and set the trigger mode for scanning.
+        /// </summary>
+        public TriggerMode TriggerMode
+        {
+            get => _triggerMode;
+            set
+            {
+                byte[] suffix = new byte[] { 0x8A, 0x08, 0xFE, 0x95 };
+
+                switch (value)
+                {
+                    case TriggerMode.Host:
+                        // this is the default on the suffix
+                        break;
+
+                    case TriggerMode.KeyHold:
+                        suffix[_param1Index] = 0x00;
+                        suffix[_param2Index] = 0x9D;
+                        break;
+
+                    case TriggerMode.KeyPress:
+                        suffix[_param1Index] = 0x02;
+                        suffix[_param2Index] = 0x9B;
+                        break;
+
+                    case TriggerMode.Continuous:
+                        suffix[_param1Index] = 0x04;
+                        suffix[_param2Index] = 0x99;
+                        break;
+
+                    case TriggerMode.Autosensing:
+                        suffix[_param1Index] = 0x09;
+                        suffix[_param2Index] = 0x94;
+                        break;
+                }
+
+                SendCommandType07(
+                    suffix,
+                    true);
+
+                // store mode
+                _triggerMode = value;
+            }
+        }
+
+        /// <summary>
+        /// Set the buzzer volume intensity.
+        /// </summary>
+        public BuzzerVolume BuzzerVolume
+        {
+            set
+            {
+                byte[] suffix = new byte[] { 0x8C, 0x00, 0xFE, 0x9B };
+
+                switch (value)
+                {
+                    case BuzzerVolume.High:
+                        // this is the default on the suffix
+                        break;
+
+                    case BuzzerVolume.Medium:
+                        suffix[_param1Index] = 0x01;
+                        suffix[_param2Index] = 0x9A;
+                        break;
+
+                    case BuzzerVolume.Low:
+                        suffix[_param1Index] = 0x02;
+                        suffix[_param2Index] = 0x99;
+                        break;
+                }
+
+                SendCommandType07(suffix);
+            }
+        }
+
+        /// <summary>
+        /// <see langword="true"/> to enable all beeps. <see langword="false"/> to disable all beeps.
+        /// </summary>
+        public bool EnableBeep
```

Comment:
same, why not having it read property as well?
And the description should start with "Sets to"

Reply:
Having a hard time coming up with a sentence to describe this that starts with "Sets".... suggestions?

Reply:
You don't like my suggestion? :-)

Reply:
I do. And I've put it there... 😯

Reply:
I did the comment before your last push ;-)